CRM/Contribute/BAO/Contribution.php

```diff
@@ -5559,6 +5502,72 @@ protected static function getContributionPaymentLinks($id, $balance, $contributi
     return $actionLinks;
   }
 
+  /**
+   * @param array $contactIDs
+   *
+   * @return string
+   */
+  public static function getAnnualQuery($contactIDs) {
+    $contactIDs = implode(',', $contactIDs);
+    $config = CRM_Core_Config::singleton();
+    $currentMonth = date('m');
+    $currentDay = date('d');
+    if ((int ) $config->fiscalYearStart['M'] > $currentMonth ||
+      ((int ) $config->fiscalYearStart['M'] == $currentMonth &&
+        (int ) $config->fiscalYearStart['d'] > $currentDay
+      )
+    ) {
+      $year = date('Y') - 1;
+    }
+    else {
+      $year = date('Y');
+    }
+    $nextYear = $year + 1;
+
+    if ($config->fiscalYearStart) {
+      $newFiscalYearStart = $config->fiscalYearStart;
+      if ($newFiscalYearStart['M'] < 10) {
```

Comment:
The significance of 10 is unclear. What does 10 represent? Can this be a constant or pseudoconstant for clarity?